openclaw_hooks/nima-recall-live/build_embedding_index.py

```
@@ -0,0 +1,184 @@
+    min_timestamp = int((datetime.now() - timedelta(days=180)).timestamp() * 1000)
+    
+    if incremental and existing_matrix is not None:
+        # Only get new embeddings
+        cursor = db.execute("""
+            SELECT id, turn_id, embedding, timestamp
+            FROM memory_nodes
+            WHERE embedding IS NOT NULL
+            AND timestamp >= ?
+            AND id NOT IN ({})
+            ORDER BY timestamp DESC
+        """.format(','.join(map(str, existing_ids)) if existing_ids else '0'), (min_timestamp,))
```

Ruff (0.15.0)

[error] Possible SQL injection vector through string-based query construction

(S608)